File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    // TC - 0(N)
    // SC - 0(N)
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        int n = nums.size();

        if(n < 2) {
            return {};
        }

        for(int i = 0; i < n ; i++) {
            if(umap.find(target - nums[i]) != umap.end()) {
                return {i, umap[target - nums[i]]};
            }
            umap[nums[i]] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer>  mp = new HashMap<>();

        for(int i = 0; i< n; i++)  {
            int temp = target - nums[i];
            if(mp.containsKey(temp)) {
                return new int[]{i, mp.get(temp)};
            }
            mp.put(nums[i], i);
        }
        return new int[]{};
    }
}